src/job_cache/eviction_policy.cpp

```diff
@@ -364,6 +420,32 @@ LRUEvictionPolicy::LRUEvictionPolicy(uint64_t max, uint64_t low)
 
 LRUEvictionPolicy::~LRUEvictionPolicy() {}
 
+void TTLEvictionPolicy::init(const std::string& cache_dir) {
+  std::shared_ptr<job_cache::Database> db = std::make_unique<job_cache::Database>(cache_dir);
+  impl = std::make_unique<TTLEvictionPolicyImpl>(cache_dir, db);
+  impl->cleanup(seconds_to_live);
+
+  // To keep this thread alive, we assign it to a static thread object.
+  // This starts the collection but if the programs ends so too will this thread.
+  gc_thread = std::thread(garbage_collect_orphan_folders, db);
+}
+
+void TTLEvictionPolicy::read(int job_id) {}
+
+void TTLEvictionPolicy::write(int job_id) {
+    static time_t time_of_last_cleanup = 0;
+    time_t current = time(nullptr);
+    // We spend a bit of time seeing if we even need to clean anything up
+    if (current - time_of_last_cleanup > 5) {
```

Comment:
The `5` here should probably be a `resolution` variable in `EvictionConfig`

Reply:
It's a lot of extra plumbing. Do you mind if we skip this for now?

Reply:
yeah thats fine